src/strategy/GGVStrategy.sol

```diff
@@ -9,132 +12,228 @@ import {WithdrawalQueue} from "src/WithdrawalQueue.sol";
+        uint256 wstethToBurn = WSTETH.getWstETHByStETH(_stethAmount);
+        requestId = requestExitByWsteth(wstethToBurn, _params);
     }
 
-    /// @notice Previews the amount of stETH shares that can be withdrawn by a given amount of GGV shares
-    /// @param _user The user to preview the amount of stETH shares for
-    /// @param _ggvShares The amount of GGV shares to preview the amount of stETH shares for
-    /// @param _params The parameters for the withdrawal
-    /// @return stethShares The amount of stETH shares that can be withdrawn
-    function previewStethSharesByGGV(address _user, uint256 _ggvShares, bytes calldata _params)
+    /**
+     * @notice Previews the amount of wstETH that can be withdrawn by a given amount of GGV shares
+     * @param _user The user to preview the amount of wstETH for
+     * @param _ggvShares The amount of GGV shares to preview the amount of wstETH for
+     * @param _params The parameters for the withdrawal
+     * @return wsteth The amount of wstETH that can be withdrawn
+     */
+    function previewWstethByGGV(address _user, uint256 _ggvShares, bytes calldata _params)
         external
         view
-        returns (uint256 stethShares)
+        returns (uint256 wsteth)
     {
         address callForwarder = getStrategyCallForwarderAddress(_user);
 
-        GGVParams memory params = abi.decode(_params, (GGVParams));
+        GGVParamsRequestExit memory params = abi.decode(_params, (GGVParamsRequestExit));
 
         IERC20 boringVault = IERC20(TELLER.vault());
         uint256 totalGGV = boringVault.balanceOf(callForwarder);
 
         if (totalGGV == 0) return 0;
         if (_ggvShares > totalGGV) revert InvalidGGVAmount();
 
-        uint256 totalStethSharesFromGgv =
-            BORING_QUEUE.previewAssetsOut(address(WSTETH), uint128(totalGGV), params.discount);
-        stethShares = Math.mulDiv(_ggvShares, totalStethSharesFromGgv, totalGGV);
+        uint256 totalWstethFromGgv = BORING_QUEUE.previewAssetsOut(address(WSTETH), uint128(totalGGV), params.discount);
+        wsteth = Math.mulDiv(_ggvShares, totalWstethFromGgv, totalGGV);
     }
 
-    /// @notice Requests a withdrawal of ggv shares from the strategy
-    /// @param _stethSharesToBurn The amount of steth shares to burn
-    /// @param _params The parameters for the withdrawal
-    /// @return requestId The request id
-    function requestExitByStethShares(uint256 _stethSharesToBurn, bytes calldata _params)
-        public
-        returns (bytes32 requestId)
-    {
-        GGVParams memory params = abi.decode(_params, (GGVParams));
+    /**
+     * @notice Requests a withdrawal of ggv shares from the strategy
+     * @param _wstethToBurn The amount of wsteth to burn
+     * @param _params The parameters for the withdrawal
+     * @return requestId The request id
+     */
+    function requestExitByWsteth(uint256 _wstethToBurn, bytes calldata _params) public returns (bytes32 requestId) {
```

Comment:
`_wstethToBurn` is nor burned, so probably just `wsteth` or `wstethToExit` is better